contrib/lib/src/databases.rs

```diff
@@ -0,0 +1,209 @@
+pub extern crate r2d2;
+
+use std::collections::BTreeMap;
+use std::fmt::{self, Display, Formatter};
+use std::marker::{Send, Sized};
+
+use rocket::config::{self, Value};
+
+pub use rocket_contrib_codegen::database;
+
+use self::r2d2::ManageConnection;
+
+#[cfg(any(feature = "diesel_sqlite_pool", feature = "diesel_postgres_pool", feature = "diesel_mysql_pool"))]
+pub extern crate diesel;
```

Comment:
I was thinking it might make sense not to reexport Diesel or any of these crates, for that matter, if possible.

You may have had this notion already, but I'm just coming around to it. It makes things simpler since the `todo` example still required using Diesel for its macros. I couldn't find any easy way to reexport them from `rocket_contrib` and decided it's probably not what we want to do.